modules/common/src/main/ts/api/ErrorExtractor.ts

```diff
@@ -0,0 +1,109 @@
+import * as Differ from './Differ';
+import * as TestError from './TestError';
+import * as LoggedError from './LoggedError';
+
+type LoggedError = LoggedError.LoggedError;
+
+type TestError = TestError.TestError;
+type PprintAssertionError = TestError.PprintAssertionError;
+type HtmlDiffAssertionError = TestError.HtmlDiffAssertionError;
+type AssertionError = TestError.AssertionError;
+
+export interface BasicErrorData {
+  readonly type: string;
+  readonly message: string;
+  readonly diff?: {
+    readonly expected: string;
+    readonly actual: string;
+    readonly comparison: string;
+  };
+}
+
+export interface ErrorData extends BasicErrorData {
+  readonly logs?: string;
+  readonly stack?: string;
+}
+
+const stringify = (e: any) => {
+  if (e === undefined) {
+    return 'undefined';
+  } else if (typeof e === 'string') {
+    return e;
+  } else {
+    return JSON.stringify(e);
+  }
+};
+
+const extractError = (err?: LoggedError): TestError =>
+  err === undefined ? new Error('no error given') : err;
+
+const extractStack = (e: TestError): string => {
+  if (e.stack) {
```

Comment:
```suggestion
  if (e.stack !== undefined) {
```

Reply:
That would be a breaking change, so we can't do that. It's possible the stack could be any falsy variant (it could be undefined, could be an empty string, etc...). We don't have our nice Katamari APIs here so we still have to do this occasionally (it was also like this in the previous code before I moved it).